package.json

```diff
@@ -31,6 +31,7 @@
     "@angular/core": "^4.0.0",
     "@angular/forms": "^4.0.0",
     "@angular/http": "^4.0.0",
+    "@angular/router": "^4.0.0",
```

Comment:
I don't want to introduce a dependency on `@angular/router`, since people can use different routers. 

Why wouldn't this approach be the same as the dialog, which uses `Location`?

Reply:
@jelbourn, I just encountered this issue when I migrated from `HttpModule` to the new `HttpClientModule`. I decided to remove the dependency (`@angular/http`) from the `package.json` file since I thought that I no longer need it. To my surprise this broke my build 😢.